parsl/config.py

```diff
@@ -132,6 +135,7 @@ def __init__(self,
         self.strategy_period = strategy_period
         self.max_idletime = max_idletime
         self.usage_tracking = usage_tracking
+        self._validate_usage_tracking()
```

Comment:
Ben had the suggestion in an earlier comment of having usage_tracking parameter default to `None` meaning unspecified in the same way that `PARSL_TRACKING` env variable can also represent unspecified by being not set. 

I can change `None` to be a constant greater than `3` to represent the unspecified state. Making it `-1`, I think would be confusing when selecting the minimum level if usage tracking is set both in the config and the environment

Reply:
@khk-globus `None` represents "no preference specified here" - it doesn't have any concrete integer tracking level equivalent, but instead indicates to the defaulting code later on that when choosing between: the `Config` specified tracking level, the `$PARSL_TRACKING` specified tracking level and the default tracking level, this config has no preference in that choice.

The integer representation comes along in code that is also aware of `$PARSL_TRACKING` in parsl/usage_tracking/usage.py to evaluate that "what's every bit of preference combined?".

So the value in `Config` is "user tracking preference" (?) that can be "no preference" or "one of these levels", the same possibilities as expressed in the `PARSL_TRACKING` tracking preference environment variable  - vs the actual tracking level later, which must be just one of the concrete levels synthesised from all preferences.

It's probably a bit over-engineered to even have these two different ways of specifying tracking level at all, given the last 6 years neglect of the usage project, and the ridiculous discrete maths we're ending up using to model and resolve this...

... but that's what posterity has delivered us (see issue #34 and related work in 2018).

Later on, usage_tracking makes use of an approximation of positive infinity (in place of None) to evaluate the minimum value of the ordered set 0<1<2<3<None... potentially this code could make that substitution right here, but it feels like a weird user experience when logging out configs: the requested tracking level is not infinity. the infinity is there because it's the 0-element of the `min` monoid, not because it's a desired tracking level.